rivet.sh

```diff
@@ -23,109 +24,42 @@ prepend_path:
-                    --disable-doxygen                                   \
-                    --with-yoda="$YODA_ROOT"                            \
-                    --with-hepmc3="$HEPMC3_ROOT"                        \
-                    --with-fastjet="$FASTJET_ROOT"                      \
-                    LDFLAGS="${LOCAL_LDFLAGS}"                          \
-                    CYTHON="$PYTHON_MODULES_ROOT/bin/cython"
-  ;;
+      EXTRA_LDFLAGS="-Wl,--no-as-needed"
+    ;;
 esac
 
-# Fix-up rivet-build to include LDFLAGS - needed _before_ bulding 
-# After 3.1.9 this will not be needed. 
-SED_EXPR="s|^myldflags=\"\(.*\)\"|myldflags=\"\1 ${LOCAL_LDFLAGS}\"|"
-sed -e "${SED_EXPR}" bin/rivet-build > bin/rivet-build.0
-mv bin/rivet-build bin/rivet-build.orig
-mv bin/rivet-build.0 bin/rivet-build
-chmod 0755 bin/rivet-build
+./configure --prefix="$INSTALLROOT"                             \
+            --disable-silent-rules                              \
+            --disable-doxygen                                   \
+            --with-yoda="$YODA_ROOT"                            \
+            --with-hepmc3="$HEPMC3_ROOT"                        \
+            --with-fastjet="$FASTJET_ROOT"                      \
+            LDFLAGS="${CGAL_ROOT:+-L${CGAL_ROOT}/lib} ${GMP_ROOT:+-L${GMP_ROOT}/lib} ${HDF5_ROOT:+-L${HDF5_ROOT}/lib} ${EXTRA_LDFLAGS}" \
+            CPPFLAGS="${CGAL_ROOT:+-I${CGAL_ROOT}/include} ${GMP_ROOT:+-I${GMP_ROOT}/include} ${HDF5_ROOT:+-I${HDF5_ROOT}/include}" \
+            CYTHON="$PYTHON_MODULES_ROOT/bin/cython"
+
 # Remove -L/usr/lib from pyext/build.py 
-sed -e 's,-L/usr/lib[^ /"]*,,g' pyext/build.py > pyext/build.py.0
-mv pyext/build.py pyext/build.py.orig
-mv pyext/build.py.0 pyext/build.py
-# Make pyext/build.py executable
-chmod 0755 pyext/build.py
+sed -i.bak -e 's,-L/usr/lib[^ /"]*,,g' pyext/build.py
 # Now build 
 make ${JOBS+-j $JOBS}
 make install
-)
 
 # Remove libRivet.la
 rm -f "$INSTALLROOT"/lib/libRivet.la
 
-# Install shell-script fragment to set-up variables 
-cp rivet_3rdparty.sh "$INSTALLROOT"/etc/rivet_3rdparty.sh
-
-# Dependencies relocation: rely on runtime environment.  That is,
-# specific paths in the generated script are replaced by expansions of
-# the relevant environment variables.
-SED_EXPR="s!x!x!"  # noop
-for P in $REQUIRES $BUILD_REQUIRES; do
-  UPPER=$(echo "$P" | tr '[:lower:]' '[:upper:]' | tr '-' '_')
-  EXPAND=$(eval echo \$"${UPPER}"_ROOT)
-  if [ -z "$EXPAND" ]; then
-      echo "Environment variable ${UPPER}_ROOT not set"
-  fi
-  [[ $EXPAND ]] || continue
-  echo "Environment says ${UPPER} is at ${EXPAND}"
-  SED_EXPR="$SED_EXPR; s!$EXPAND!\$${UPPER}_ROOT!g"
-done
-
 # Create line to source 3rdparty.sh to be inserted into 
 # rivet-config and rivet-build 
 cat << EOF > source3rd
-test -f \$prefix/etc/rivet_3rdparty.sh && source \$prefix/etc/rivet_3rdparty.sh
+WORK_DIR=${WORK_DIR:-$INSTALLROOT/../../../} source $INSTALLROOT/etc/profile.d/init.sh
```

Comment:
```suggestion
WORK_DIR=$(dirname $(type rivet-build))/../../../../ source $(dirname $(type rivet-build))/../etc/profile.d/init.sh 
```

Reply:
/home/mgiacalo/docker-alice/sw/ubuntu2404_x86-64/Rivet/rivet-4.1.0-local2/bin/rivet-build: line 133: /../etc/profile.d/init.sh: No such file or directory
This doesn't seem to work unfortunately

Reply:
let's have a look later today